Lombiq.DataTables/Tests/Lombiq.DataTables.Tests/UnitTests/Services/ExportTests.cs

```diff
@@ -246,4 +240,50 @@ public static IEnumerable<object[]> Data()
             0,
         };
     }
+
+    // Sometimes a font is available, however, it's corrupted or missing a table (for example, this can happen on
+    // GitHub-hosted runners). We can't check directly if a font is missing a table or corrupted, but we can try
+    // other fonts if this happens.
+    private static async Task<Stream> TryExportWithFallbackFontsAsync(
+        ExcelDataTableExportService service,
+        IDataTableDataProvider provider,
+        DataTableDataRequest request,
+        IDictionary<int, string> customNumberFormat)
+    {
+        var fontFamilies = SystemFonts.Collection.Families.ToArray();
+
+        var maxAttempts = Math.Min(3, fontFamilies.Length);
+        var fontIndex = 0;
+
+        while (fontIndex < maxAttempts)
+        {
+            var fallbackFont = fontFamilies[fontIndex].Name;
+
+            LoadOptions.DefaultGraphicEngine = new DefaultGraphicEngine(fallbackFont);
+
+            try
+            {
+                return await service.ExportAsync(provider, request, customNumberFormat: customNumberFormat);
+            }
+            catch (MissingFontTableException missingFontTableException)
+            {
+                DebugHelper.WriteLineTimestamped(
+                    $"Attempt {(fontIndex + 1).ToTechnicalString()} of exporting the data table with the font " +
+                    $"{fallbackFont} failed with the MissingFontTableException: {missingFontTableException.Message}.");
+
+                if (fontIndex + 1 < maxAttempts)
+                {
+                    DebugHelper.WriteLineTimestamped("Retrying with a different font...");
+                }
+                else
+                {
+                    throw missingFontTableException;
```

Comment:
Does this throw the same exception on purpose instead of re-throwing it? It is generally good advice to avoid this, because you lose the whole stacktrace, but that might be desired here - just asking to make sure.

Otherwise, this is how we would re-throw:

```suggestion
                    throw;
```

Reply:
I want to pass the stack trace, thanks!